traffic_portal/app/src/common/modules/table/types/TableTypesController.js

```diff
@@ -33,15 +37,34 @@ var TableTypesController = function(types, $scope, $state, locationUtils) {
         $state.reload(); // reloads all the resolves for the view
     };
 
+    $scope.toggleVisibility = function(colName) {
+        const col = table.column(colName + ':name');
+        col.visible(!col.visible());
+        table.rows().invalidate().draw();
+    };
+
     angular.element(document).ready(function () {
-        $('#typesTable').dataTable({
-            "aLengthMenu": [[25, 50, 100, -1], [25, 50, 100, "All"]],
-            "iDisplayLength": 25,
-            "aaSorting": []
-        });
+        table = $('#typesTable').DataTable({
+                "lengthMenu": [[25, 50, 100, -1], [25, 50, 100, "All"]],
+                "iDisplayLength": 25,
+                "aaSorting": [],
+                "columns": [
+                    { "name": "name", "visible": true, "searchable": true },
+                    { "name": "description", "visible": true, "searchable": true },
+                    { "name": "lastUpdated", "visible": false, "searchable": false },
+                    { "name": "useInTable", "visible": true, "searchable": true }
+                ],
+                "colReorder": {
+                    realtime: false
+                },
+                "initComplete": function(settings, json) {
+                    // need to create the show/hide column checkboxes and bind to the current visibility
+                    $scope.columns = JSON.parse(localStorage.getItem('DataTables_typesTable_/')).columns;
```

Comment:
You should still handle this error, because an error thrown by `JSON.parse` doesn't necessarily mean that `localStorage` isn't working - the locally stored data could have been corrupted somehow etc. Stack traces from an angular compile cycle are very hard to follow, so even if all you do with the error is log it to the console it could aid in debugging.

Reply:
handle with a try/catch?

Reply:
Yeah, a simple

```javascript
try {
    doThing()
} catch e {
    console.error("Failed to load columns from localStorage:", e)
    // optional fallback column assignment here (not sure if that's necessary)
}
```

would be fine IMO. Esp. since that error could propagate upward and mess up the whole UI, not sure what that would look like.

Reply:
actually, failure to parse will just result in an empty dropdown

![image](https://user-images.githubusercontent.com/251272/59867078-6b62a280-934a-11e9-8a58-a3e2ea51232d.png)

but i like the idea of providing some sort of info as to the problem.